client-js/app/elements/labrad-plot.ts

```diff
@@ -154,49 +240,129 @@ export class Plot extends polymer.Base {
     for (let row of data) {
       this.splice('data', this.data.length, 0, row);
     }
-    this.plotData(data, lastData);
+    this.plotData(data);
+
+    if (!this.haveZoomed) {
+      this.resetZoom();
+    } else {
+      this.projectGraphPositions();
+    }
   }
 
 
-  private createPlot(
-      area: HTMLElement, totWidth: number, totHeight: number): void {
+  /**
+   * Redraw the plot and attach resize to the window resize event.
+   * Fires when the component is attached to the DOM.
+   */
+  attached() {
+    this.render();
+    window.addEventListener('resize', (e) => this.resizePlot());
+  }
+
+
+  /**
+   * Resets the current zoom level to fit the data.
+   */
+  resetZoomControl() {
+    this.resetZoom();
+  }
+
+
+  /**
+   * Creates and renders the SVG and WebGL portions of the graph.
+   */
+  private render(): void {
+    if (this.isRendering) {
+      return;
+    }
+
+    this.isRendering = true;
+
+    // Initialize SVG portions of plot.
+    this.initializeSVGPlot();
+
+    // Initialize WebGL portions of plot.
+    this.initializeWebGLPlot();
+
+    // Resize the plot to be the correct dimensions.
+    this.resizePlot();
+  }
+
+
+  /**
+   * Resizes the SVG and WebGL portions of the plots.
+   */
+  private resizePlot() {
+    this.resizeSVGPlot();
+    this.resizeWebGLPlot();
+    this.projectGraphPositions();
+  }
+
+
+  /**
+   * Begins the scene rendering loop.
+   */
+  private renderScene(): void {
```

Comment:
The name and documentation here is a bit confusing because this does not just begin the render loop, this _is_ the render loop. Also, we need to shut down the render loop when this plot element is detached.